connmgr/connmgr.go

```diff
@@ -1,3 +1,9 @@
+// Package connmgr provides connection tracking and management interfaces for libp2p.
+//
+// The ConnManager interface exported from this package allows libp2p to enforce an
+// upper bound on the total number of open connections, which prevents us from
+// overwhelming routers with too many connections. Without such a mechanism, many
+// consumer routers will begin to fail and behave erratically given enough connections.
```

Comment:
Well that's not the only reason to enforce upper bound in connections. We also care about memory and resource usage from large number of connections.

Reply:
That's a good point; it's probably enough just to mention that we want to impose an upper bound. There are enough obvious motivations that we don't need to distract people by talking about choking routers.